portal-web/test/functional/com/liferay/portalweb/tests/enduseree/formsworkflow/metrics/WorkflowMetrics.testcase

```diff
@@ -98,6 +98,106 @@ definition {
 			value1 = "${key_testSLADefinitionDays}");
 	}
 
+	@description = "Apply filter for Assignee Search(LRQA-53633)"
+	@priority = "5"
+	test ApplySearchesForAllAssignees {
+
+		ProductMenu.gotoPortlet(
+			category = "Workflow",
+			panel = "Control Panel",
+			portlet = "Process Builder");
+
+		Navigator.gotoNavItem(navItem = "Configuration");
+
+		Workflow.configureWorkflow(
+			workflowDefinition = "Single Approver",
+			workflowResourceValue = "Blogs Entry");
+
+		ProductMenu.gotoPortlet(
+			category = "Content &amp; Data",
+			panel = "Site Administration",
+			portlet = "Blogs");
+
+		for (var count : list "1,2,3") {
+			BlogsNavigator.gotoAddEntry();
+
+			BlogsEntry.addEntryContent(
+				entryContent = "test content ${count}",
+				entryTitle = "Test ${count}");
+
+			Click(
+				locator1 = "Button#SUBMIT_FOR_PUBLICATION",
+				value1 = "Submit for Publication");
+		}
+
+		WorkflowMetrics.executeWorkflowScript(scriptName = "addUsersWithRoles.groovy");
+
+		UserBar.gotoDropdownItem(dropdownItem = "My Workflow Tasks");
+
+		Navigator.gotoNavTab(navTab = "Assigned to My Roles");
+
+		Workflow.assignTaskToUser(
+			assetTitle = "Test 1",
+			assetType = "Blogs Entry",
+			assigneeName = "1user user1",
+			workflowTask = "Review");
+
+		Workflow.assignTaskToUser(
+			assetTitle = "Test 2",
+			assetType = "Blogs Entry",
+			assigneeName = "2user user2",
+			workflowTask = "Review");
+
+		Workflow.assignTaskToUser(
+			assetTitle = "Test 3",
+			assetType = "Blogs Entry",
+			assigneeName = "3user user3",
+			workflowTask = "Review");
+
+		ProductMenu.gotoPortlet(
+			category = "Workflow",
+			panel = "Control Panel",
+			portlet = "Metrics");
+
+		WorkflowMetrics.goToWorkflowProcessMetrics(workflowProcessName = "Single Approver");
+
+		Click(
+			key_assigneesNumber = "3",
+			locator1 = "WorkflowMetrics#METRICS_PERFORMANCE_BY_ASSIGNEE_CARD_VIEW_ALL_ASSIGNEES_LINK");
+
+		for (var count : list "1,2,3") {
+			WorkflowMetrics.selectWorkloadByAssigneeCardFilterValue(workflowStep = "${count}Role");
+
+			Click(locator1 = "WorkflowMetrics#METRICS_WORKLOAD_BY_ASSIGNEE_SEARCH_ASSIGNEE_NAME");
+
+			WorkflowMetrics.viewWorkloadByAssigneeSingleApprover(
+				assigneeName = "${count}user user${count}",
+				onTimeTaskCount = "0",
+				overdueTaskCount = "0",
+				rowNumber = "${count}",
+				totalPendingTaskCount = "1");
+		}
+
+		WorkflowMetrics.selectWorkloadByAssigneeCardFilterValueProcessStep(workflowStep = "Review");
+
+		Click(locator1 = "WorkflowMetrics#METRICS_WORKLOAD_BY_ASSIGNEE_SEARCH_ASSIGNEE_NAME");
+
+		WorkflowMetrics.selectWorkloadByAssigneeCardFilterValueProcessStep(workflowStep = "Update");
+
+		WorkflowMetrics.searchForAssigneeName(assigneeName = "something else");
+
+		WorkflowMetrics.clearAllFilter();
+
+		for (var count : list "1,2,3") {
+			WorkflowMetrics.viewWorkloadByAssigneeSingleApprover(
+				assigneeName = "${count}user user${count}",
+				onTimeTaskCount = "0",
+				overdueTaskCount = "0",
+				rowNumber = "${count}",
```

Comment:
Hi @calvinctvo,
here you validate the table entries, right? Though, the way it is right now, as far as I understood, we are explicitly defining the rowNumber. It is an approach to validate it, but it is not scalable, unfortunately... we need to assume that we don't know exactly where the the entry will appear on the table, unless the test explicitly requires a position (sorting, for example).

So, a good approach is to make the xpath locator more robust. You can check for following-sibling, and ancestor for xpath...

For example:

Assignee name locator: //span[contains(text(),'**1user user1**')]
Overdue tasks value: (//span[contains(text(),'1user')]/ancestor::td[contains(@class,'table-title')]/following-sibling::td//a)[1]
On Time tasks value: (//span[contains(text(),'1user')]/ancestor::td[contains(@class,'table-title')]/following-sibling::td//a)[2]
Total Pending: (//span[contains(text(),'1user')]/ancestor::td[contains(@class,'table-title')]/following-sibling::td//a)[3]

Like that, we would not have to worry about the rowNumber.